onnxruntime/core/providers/shared_library/bridge.h

```diff
@@ -0,0 +1,95 @@
+namespace onnxruntime {
+
+struct IExecutionProviderFactory;
+struct ProviderHost;
+struct KernelCreateInfo;
+class CPUIDInfo;
+
+namespace logging {
+class Logger;
+}
+
+struct Provider {
+  virtual std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory(int device_id) = 0;
```

Comment:
>std::shared_ptr [](start = 10, length = 15)

Can we return this as unique_ptr and whoever needs can convert that unique_ptr to shared?